### ISSUE TITLE
Pytest for VSCode failing.

### ISSUE THREAD

**Issue by Joseph-Garvey** [palvarezc/cavendish-particle-tracks#96]
Attempting to fix test failures in PR for issue #90 but PyTest has stopped working when I've come in today.
Worked fine on Friday, suspected update but have tried the following troubleshooting steps:
1. Clearing .vscode/settings.json
2. Adding "--no-cov" to the config.
3. Reinstalling the python extension for VSCode.
4. Rolling back the python extension for VSCode to a version before last week.
5. Creating a fresh python venv.

PyTest works fine when called from the command line within the project directory. Tests run and pass, but VSCode error gives "pyTest Discovery Error" and
```powershell
2024-08-19 14:06:03.563 [info] Experiment 'pythonaa' is active
2024-08-19 14:06:03.563 [info] Experiment 'pythonCreateEnvOnPipInstall' is active
2024-08-19 14:06:03.563 [info] Experiment 'pythonRecommendTensorboardExt' is active
2024-08-19 14:06:03.563 [info] Experiment 'pythonTerminalEnvVarActivation' is active
2024-08-19 14:06:03.563 [info] Experiment 'pythonTestAdapter' is active
2024-08-19 14:06:03.563 [info] Native locator: Refresh started
2024-08-19 14:06:03.563 [info] Default formatter is set to ms-python.black-formatter for workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:03.563 [info] > pyenv which python
2024-08-19 14:06:03.563 [info] cwd: ~\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:03.570 [info] Conda environment manager found at: C:\ProgramData\anaconda3\Scripts\conda.exe
2024-08-19 14:06:03.573 [info] Python interpreter path: ~\.conda\envs\napari-env\python.exe
2024-08-19 14:06:03.592 [info] Native locator: Refresh finished in 390 ms
2024-08-19 14:06:03.758 [info] > conda info --json
2024-08-19 14:06:03.758 [info] shell: powershell
2024-08-19 14:06:04.343 [info] Discover tests for workspace name: cavendish-particle-tracks - uri: c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks\.vscode\settings.json
2024-08-19 14:06:04.343 [info] Running discovery for pytest using the new test adapter.
2024-08-19 14:06:04.344 [warning] The cwd resolves to a different path, checking if it has a symbolic link somewhere in its path.
2024-08-19 14:06:04.346 [info] All environment variables set for pytest discovery: {"ALLUSERSPROFILE":"C:\\ProgramData","APPDATA":"C:\\Users\\josep\\AppData\\Roaming","CHROME_CRASHPAD_PIPE_NAME":"\\\\.\\pipe\\crashpad_36472_HSZYTGMNHRGZRXPX","COMPUTERNAME":"LAPTOP-TLC5JM8B","ELECTRON_RUN_AS_NODE":"1","HOMEDRIVE":"C:","HOMEPATH":"\\Users\\josep","LOCALAPPDATA":"C:\\Users\\josep\\AppData\\Local","LOGONSERVER":"\\\\LAPTOP-TLC5JM8B","NUMBER_OF_PROCESSORS":"16","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","OS":"Windows_NT","PATHEXT":".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC","PROCESSOR_ARCHITECTURE":"AMD64","PROCESSOR_IDENTIFIER":"AMD64 Family 25 Model 80 Stepping 0, AuthenticAMD","PROCESSOR_LEVEL":"25","PROCESSOR_REVISION":"5000","PUBLIC":"C:\\Users\\Public","SESSIONNAME":"Console","TEMP":"C:\\Users\\josep\\AppData\\Local\\Temp","TMP":"C:\\Users\\josep\\AppData\\Local\\Temp","USERDOMAIN":"LAPTOP-TLC5JM8B","USERDOMAIN_ROAMINGPROFILE":"LAPTOP-TLC5JM8B","USERNAME":"gravye","USERPROFILE":"C:\\Users\\josep","VSCODE_AMD_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","VSCODE_CODE_CACHE_PATH":"C:\\Users\\josep\\AppData\\Roaming\\Code\\CachedData\\fee1edb8d6d72a0ddff41e5f71a671c23ed924b9","VSCODE_CRASH_REPORTER_PROCESS_TYPE":"extensionHost","VSCODE_CWD":"C:\\Users\\josep\\AppData\\Local\\Programs\\Microsoft VS Code","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_IPC_HOOK":"\\\\.\\pipe\\801452dc-1.92.2-main-sock","VSCODE_L10N_BUNDLE_LOCATION":"","VSCODE_NLS_CONFIG":"{\"userLocale\":\"en-us\",\"osLocale\":\"en-gb\",\"resolvedLanguage\":\"en\",\"defaultMessagesFile\":\"C:\\\\Users\\\\josep\\\\AppData\\\\Local\\\\Programs\\\\Microsoft VS Code\\\\resources\\\\app\\\\out\\\\nls.messages.json\",\"locale\":\"en\",\"availableLanguages\":{}}","VSCODE_PID":"36472","ACSetupSvcPort":"23210","ACSvcPort":"17532","ComSpec":"C:\\WINDOWS\\system32\\cmd.exe","CommonProgramFiles":"C:\\Program Files\\Common Files","CommonProgramFiles(x86)":"C:\\Program Files (x86)\\Common Files","CommonProgramW6432":"C:\\Program Files\\Common Files","DriverData":"C:\\Windows\\System32\\Drivers\\DriverData","OneDrive":"C:\\Users\\josep\\OneDrive - University of Cambridge","OneDriveCommercial":"C:\\Users\\josep\\OneDrive - University of Cambridge","PSModulePath":"C:\\Program Files\\WindowsPowerShell\\Modules;C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\Modules","ProgramData":"C:\\ProgramData","ProgramFiles":"C:\\Program Files","ProgramFiles(x86)":"C:\\Program Files (x86)","ProgramW6432":"C:\\Program Files","RlsSvcPort":"22112","SystemDrive":"C:","SystemRoot":"C:\\WINDOWS","windir":"C:\\WINDOWS","Path":"C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Windows\\System32\\OpenSSH\\;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\dotnet\\;C:\\Program Files\\Git\\cmd;C:\\Program Files\\NVIDIA Corporation\\NVIDIA NvDLISR;C:\\Users\\josep\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\josep\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\josep\\.dotnet\\tools","PYTHONPATH":"c:\\Users\\josep\\.vscode\\extensions\\ms-python.python-2024.12.1-win32-x64\\python_files","TEST_RUN_PIPE":"\\\\.\\pipe\\python-test-discovery-150250050de6883f25f6-sock"}
2024-08-19 14:06:04.354 [info] > C:\ProgramData\anaconda3\Scripts\conda.exe info --json
2024-08-19 14:06:04.739 [info] Discover tests for workspace name: cavendish-particle-tracks - uri: c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:04.739 [info] Running discovery for pytest using the new test adapter.
2024-08-19 14:06:04.739 [error] Test discovery already in progress, not starting a new one.
2024-08-19 14:06:04.773 [info] > C:\ProgramData\anaconda3\Scripts\conda.exe info --json
2024-08-19 14:06:04.773 [info] shell: powershell
2024-08-19 14:06:05.041 [info] Discover tests for workspace name: cavendish-particle-tracks - uri: c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:05.041 [info] Running discovery for pytest using the new test adapter.
2024-08-19 14:06:05.042 [error] Test discovery already in progress, not starting a new one.
2024-08-19 14:06:05.502 [info] Starting Pylance language server.
2024-08-19 14:06:08.127 [info] > C:\ProgramData\anaconda3\Scripts\conda.exe run -n napari-env --no-capture-output python ~\.vscode\extensions\ms-python.python-2024.12.1-win32-x64\python_files\get_output_via_markers.py ~/.vscode/extensions/ms-python.python-2024.12.1-win32-x64/python_files/printEnvVariables.py
2024-08-19 14:06:08.127 [info] shell: commandPrompt
2024-08-19 14:06:08.142 [info] > C:\ProgramData\anaconda3\condabin\conda.bat info --json
2024-08-19 14:06:08.142 [info] shell: powershell
2024-08-19 14:06:09.800 [info] > pixi --version
2024-08-19 14:06:09.801 [error] Error: spawn pixi ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn pixi',
  path: 'pixi',
  spawnargs: [ '--version' ]
}
2024-08-19 14:06:09.801 [warning] could not find a pixi interpreter for the interpreter at C:\Users\josep\.conda\envs\napari-env\python.exe
2024-08-19 14:06:09.827 [info] > python -m pytest -p vscode_pytest --collect-only ./tests --no-cov
2024-08-19 14:06:09.827 [info] cwd: ~\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:09.904 [error] Python was not found; run without arguments to install from the
2024-08-19 14:06:09.904 [error]  Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

2024-08-19 14:06:09.910 [error] Subprocess exited unsuccessfully with exit code 9009 and signal null on workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks.
2024-08-19 14:06:09.910 [error] Subprocess exited unsuccessfully with exit code 9009 and signal null on workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks. Creating and sending error discovery payload
2024-08-19 14:06:09.911 [error] pytest test discovery error for workspace:  c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks 
  
 The python test process was terminated before it could exit on its own, the process errored with: Code: 9009, Signal: null for workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:10.028 [info] > C:\ProgramData\anaconda3\Scripts\conda.exe run -n napari-env --no-capture-output python ~\.vscode\extensions\ms-python.python-2024.12.1-win32-x64\python_files\get_output_via_markers.py ~/.vscode/extensions/ms-python.python-2024.12.1-win32-x64/python_files/printEnvVariables.py
2024-08-19 14:06:10.028 [info] shell: powershell
2024-08-19 14:06:12.119 [info] > python ~/.vscode/extensions/ms-python.python-2024.12.1-win32-x64/python_files/printEnvVariables.py
2024-08-19 14:06:12.119 [info] shell: powershell
2024-08-19 14:06:12.386 [info] Setting environment variable CONDA_DEFAULT_ENV in collection to napari-env {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.386 [info] Setting environment variable CONDA_EXE in collection to C:\ProgramData\anaconda3\Scripts\conda.exe {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.386 [info] Setting environment variable CONDA_PREFIX in collection to C:\Users\josep\.conda\envs\napari-env {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.386 [info] Setting environment variable CONDA_PROMPT_MODIFIER in collection to (napari-env)  {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.386 [info] Setting environment variable CONDA_PYTHON_EXE in collection to C:\ProgramData\anaconda3\python.exe {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable CONDA_SHLVL in collection to 1 {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Prepending environment variable PATH in collection to C:\Users\josep\.conda\envs\napari-env;C:\Users\josep\.conda\envs\napari-env\Library\mingw-w64\bin;C:\Users\josep\.conda\envs\napari-env\Library\usr\bin;C:\Users\josep\.conda\envs\napari-env\Library\bin;C:\Users\josep\.conda\envs\napari-env\Scripts;C:\Users\josep\.conda\envs\napari-env\bin;C:\ProgramData\anaconda3\condabin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0;C:\WINDOWS\System32\OpenSSH;C:\Program Files\dotnet;C:\Program Files\Git\cmd;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Users\josep\AppData\Local\Microsoft\WindowsApps;C:\Users\josep\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\josep\.dotnet\tools; {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable PATHEXT in collection to .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable PROMPT in collection to (napari-env) $P$G {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable PSMODULEPATH in collection to C:\Users\josep\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable PYTHONIOENCODING in collection to utf-8 {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable PYTHONUNBUFFERED in collection to 1 {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable SSL_CERT_DIR in collection to C:\Users\josep\.conda\envs\napari-env\Library\ssl\certs {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable SSL_CERT_FILE in collection to C:\Users\josep\.conda\envs\napari-env\Library\ssl\cacert.pem {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable CONDA_ROOT in collection to C:\ProgramData\anaconda3 {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable XML_CATALOG_FILES in collection to file:///C:/Users/josep/.conda/envs/napari-env/etc/xml/catalog {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable _CONDA_OLD_CHCP in collection to 850 {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable __CONDA_OPENSSL_CERT_DIR_SET in collection to "1" {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Setting environment variable __CONDA_OPENSSL_CERT_FILE_SET in collection to "1" {"applyAtShellIntegration":true,"applyAtProcessCreation":true}
2024-08-19 14:06:12.387 [info] Send text to terminal: & python c:\Users\josep\.vscode\extensions\ms-python.python-2024.12.1-win32-x64\python_files\printEnvVariablesToFile.py c:\Users\josep\.vscode\extensions\ms-python.python-2024.12.1-win32-x64\python_files\deactivate\powershell\envVars.txt
2024-08-19 14:06:21.572 [info] Discover tests for workspace name: cavendish-particle-tracks - uri: c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks\.vscode\settings.json
2024-08-19 14:06:21.572 [info] Running discovery for pytest using the new test adapter.
2024-08-19 14:06:21.573 [warning] The cwd resolves to a different path, checking if it has a symbolic link somewhere in its path.
2024-08-19 14:06:21.573 [info] All environment variables set for pytest discovery: {"ALLUSERSPROFILE":"C:\\ProgramData","APPDATA":"C:\\Users\\josep\\AppData\\Roaming","CHROME_CRASHPAD_PIPE_NAME":"\\\\.\\pipe\\crashpad_36472_HSZYTGMNHRGZRXPX","COMPUTERNAME":"LAPTOP-TLC5JM8B","ELECTRON_RUN_AS_NODE":"1","HOMEDRIVE":"C:","HOMEPATH":"\\Users\\josep","LOCALAPPDATA":"C:\\Users\\josep\\AppData\\Local","LOGONSERVER":"\\\\LAPTOP-TLC5JM8B","NUMBER_OF_PROCESSORS":"16","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","OS":"Windows_NT","PATHEXT":".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC","PROCESSOR_ARCHITECTURE":"AMD64","PROCESSOR_IDENTIFIER":"AMD64 Family 25 Model 80 Stepping 0, AuthenticAMD","PROCESSOR_LEVEL":"25","PROCESSOR_REVISION":"5000","PUBLIC":"C:\\Users\\Public","SESSIONNAME":"Console","TEMP":"C:\\Users\\josep\\AppData\\Local\\Temp","TMP":"C:\\Users\\josep\\AppData\\Local\\Temp","USERDOMAIN":"LAPTOP-TLC5JM8B","USERDOMAIN_ROAMINGPROFILE":"LAPTOP-TLC5JM8B","USERNAME":"gravye","USERPROFILE":"C:\\Users\\josep","VSCODE_AMD_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","VSCODE_CODE_CACHE_PATH":"C:\\Users\\josep\\AppData\\Roaming\\Code\\CachedData\\fee1edb8d6d72a0ddff41e5f71a671c23ed924b9","VSCODE_CRASH_REPORTER_PROCESS_TYPE":"extensionHost","VSCODE_CWD":"C:\\Users\\josep\\AppData\\Local\\Programs\\Microsoft VS Code","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_IPC_HOOK":"\\\\.\\pipe\\801452dc-1.92.2-main-sock","VSCODE_L10N_BUNDLE_LOCATION":"","VSCODE_NLS_CONFIG":"{\"userLocale\":\"en-us\",\"osLocale\":\"en-gb\",\"resolvedLanguage\":\"en\",\"defaultMessagesFile\":\"C:\\\\Users\\\\josep\\\\AppData\\\\Local\\\\Programs\\\\Microsoft VS Code\\\\resources\\\\app\\\\out\\\\nls.messages.json\",\"locale\":\"en\",\"availableLanguages\":{}}","VSCODE_PID":"36472","ACSetupSvcPort":"23210","ACSvcPort":"17532","ComSpec":"C:\\WINDOWS\\system32\\cmd.exe","CommonProgramFiles":"C:\\Program Files\\Common Files","CommonProgramFiles(x86)":"C:\\Program Files (x86)\\Common Files","CommonProgramW6432":"C:\\Program Files\\Common Files","DriverData":"C:\\Windows\\System32\\Drivers\\DriverData","OneDrive":"C:\\Users\\josep\\OneDrive - University of Cambridge","OneDriveCommercial":"C:\\Users\\josep\\OneDrive - University of Cambridge","PSModulePath":"C:\\Program Files\\WindowsPowerShell\\Modules;C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\Modules","ProgramData":"C:\\ProgramData","ProgramFiles":"C:\\Program Files","ProgramFiles(x86)":"C:\\Program Files (x86)","ProgramW6432":"C:\\Program Files","RlsSvcPort":"22112","SystemDrive":"C:","SystemRoot":"C:\\WINDOWS","windir":"C:\\WINDOWS","Path":"C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Windows\\System32\\OpenSSH\\;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\dotnet\\;C:\\Program Files\\Git\\cmd;C:\\Program Files\\NVIDIA Corporation\\NVIDIA NvDLISR;C:\\Users\\josep\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\josep\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\josep\\.dotnet\\tools","PYTHONPATH":"c:\\Users\\josep\\.vscode\\extensions\\ms-python.python-2024.12.1-win32-x64\\python_files","TEST_RUN_PIPE":"\\\\.\\pipe\\python-test-discovery-e2275f3f664681c9ffc7-sock"}
2024-08-19 14:06:21.574 [warning] could not find a pixi interpreter for the interpreter at C:\Users\josep\.conda\envs\napari-env\python.exe
2024-08-19 14:06:21.592 [info] > python -m pytest -p vscode_pytest --collect-only tests --no-cov
2024-08-19 14:06:21.592 [info] cwd: ~\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:21.657 [error] Python was not found; run without arguments to install from t
2024-08-19 14:06:21.657 [error] he Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

2024-08-19 14:06:21.663 [error] Subprocess exited unsuccessfully with exit code 9009 and signal null on workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks.
2024-08-19 14:06:21.663 [error] Subprocess exited unsuccessfully with exit code 9009 and signal null on workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks. Creating and sending error discovery payload
2024-08-19 14:06:21.663 [error] pytest test discovery error for workspace:  c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks 
  
 The python test process was terminated before it could exit on its own, the process errored with: Code: 9009, Signal: null for workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:22.278 [info] Discover tests for workspace name: cavendish-particle-tracks - uri: c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:22.278 [info] Running discovery for pytest using the new test adapter.
2024-08-19 14:06:22.279 [warning] The cwd resolves to a different path, checking if it has a symbolic link somewhere in its path.
2024-08-19 14:06:22.279 [info] All environment variables set for pytest discovery: {"ALLUSERSPROFILE":"C:\\ProgramData","APPDATA":"C:\\Users\\josep\\AppData\\Roaming","CHROME_CRASHPAD_PIPE_NAME":"\\\\.\\pipe\\crashpad_36472_HSZYTGMNHRGZRXPX","COMPUTERNAME":"LAPTOP-TLC5JM8B","ELECTRON_RUN_AS_NODE":"1","HOMEDRIVE":"C:","HOMEPATH":"\\Users\\josep","LOCALAPPDATA":"C:\\Users\\josep\\AppData\\Local","LOGONSERVER":"\\\\LAPTOP-TLC5JM8B","NUMBER_OF_PROCESSORS":"16","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","OS":"Windows_NT","PATHEXT":".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC","PROCESSOR_ARCHITECTURE":"AMD64","PROCESSOR_IDENTIFIER":"AMD64 Family 25 Model 80 Stepping 0, AuthenticAMD","PROCESSOR_LEVEL":"25","PROCESSOR_REVISION":"5000","PUBLIC":"C:\\Users\\Public","SESSIONNAME":"Console","TEMP":"C:\\Users\\josep\\AppData\\Local\\Temp","TMP":"C:\\Users\\josep\\AppData\\Local\\Temp","USERDOMAIN":"LAPTOP-TLC5JM8B","USERDOMAIN_ROAMINGPROFILE":"LAPTOP-TLC5JM8B","USERNAME":"gravye","USERPROFILE":"C:\\Users\\josep","VSCODE_AMD_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","VSCODE_CODE_CACHE_PATH":"C:\\Users\\josep\\AppData\\Roaming\\Code\\CachedData\\fee1edb8d6d72a0ddff41e5f71a671c23ed924b9","VSCODE_CRASH_REPORTER_PROCESS_TYPE":"extensionHost","VSCODE_CWD":"C:\\Users\\josep\\AppData\\Local\\Programs\\Microsoft VS Code","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_IPC_HOOK":"\\\\.\\pipe\\801452dc-1.92.2-main-sock","VSCODE_L10N_BUNDLE_LOCATION":"","VSCODE_NLS_CONFIG":"{\"userLocale\":\"en-us\",\"osLocale\":\"en-gb\",\"resolvedLanguage\":\"en\",\"defaultMessagesFile\":\"C:\\\\Users\\\\josep\\\\AppData\\\\Local\\\\Programs\\\\Microsoft VS Code\\\\resources\\\\app\\\\out\\\\nls.messages.json\",\"locale\":\"en\",\"availableLanguages\":{}}","VSCODE_PID":"36472","ACSetupSvcPort":"23210","ACSvcPort":"17532","ComSpec":"C:\\WINDOWS\\system32\\cmd.exe","CommonProgramFiles":"C:\\Program Files\\Common Files","CommonProgramFiles(x86)":"C:\\Program Files (x86)\\Common Files","CommonProgramW6432":"C:\\Program Files\\Common Files","DriverData":"C:\\Windows\\System32\\Drivers\\DriverData","OneDrive":"C:\\Users\\josep\\OneDrive - University of Cambridge","OneDriveCommercial":"C:\\Users\\josep\\OneDrive - University of Cambridge","PSModulePath":"C:\\Program Files\\WindowsPowerShell\\Modules;C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\Modules","ProgramData":"C:\\ProgramData","ProgramFiles":"C:\\Program Files","ProgramFiles(x86)":"C:\\Program Files (x86)","ProgramW6432":"C:\\Program Files","RlsSvcPort":"22112","SystemDrive":"C:","SystemRoot":"C:\\WINDOWS","windir":"C:\\WINDOWS","Path":"C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Windows\\System32\\OpenSSH\\;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\dotnet\\;C:\\Program Files\\Git\\cmd;C:\\Program Files\\NVIDIA Corporation\\NVIDIA NvDLISR;C:\\Users\\josep\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\josep\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\josep\\.dotnet\\tools","PYTHONPATH":"c:\\Users\\josep\\.vscode\\extensions\\ms-python.python-2024.12.1-win32-x64\\python_files","TEST_RUN_PIPE":"\\\\.\\pipe\\python-test-discovery-ffff162c622e334ef254-sock"}
2024-08-19 14:06:22.280 [warning] could not find a pixi interpreter for the interpreter at C:\Users\josep\.conda\envs\napari-env\python.exe
2024-08-19 14:06:22.299 [info] > python -m pytest -p vscode_pytest --collect-only tests --no-cov
2024-08-19 14:06:22.299 [info] cwd: ~\Desktop\Particle Tracks\cavendish-particle-tracks
2024-08-19 14:06:22.367 [error] Python was not found; run without arguments to install from the Micr
2024-08-19 14:06:22.367 [error] osoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

2024-08-19 14:06:22.373 [error] Subprocess exited unsuccessfully with exit code 9009 and signal null on workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks.
2024-08-19 14:06:22.373 [error] Subprocess exited unsuccessfully with exit code 9009 and signal null on workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks. Creating and sending error discovery payload
2024-08-19 14:06:22.373 [error] pytest test discovery error for workspace:  c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks 
  
 The python test process was terminated before it could exit on its own, the process errored with: Code: 9009, Signal: null for workspace c:\Users\josep\Desktop\Particle Tracks\cavendish-particle-tracks
`
```

**Comment by Joseph-Garvey** [comment]
Problem solved.

**Comment by Joseph-Garvey** [comment]
Solutions tested:
My first suspicion was an update causing issues, hence the first two items that I tried:
1. Reinstalling the python extension for VSCode.
2. Rolling back the python extension for VSCode to a version before last week.
3. Clearing .vscode/settings.json
4. Adding "--no-cov" to the config.
5. Creating a fresh python venv.
6. Reinstalling VSCode
7. Reinstalling Python

Turns out it was an update that broke VSCode. Had to roll back 2 months to get it working again.
I suspect that auto-update must not have run for quite a while on my PC, hence the need to rollback so far.
Have tried updating again and it immediately breaks, so pretty much rules out other causes. Will file a bug report.
In other news the Napari bug I discovered has now been fixed in 0.5.2
